route/build/cpl/RtmMod.F90

```diff
@@ -118,16 +119,30 @@ SUBROUTINE route_ini(rof_active,flood_active)
     end if
 
     !-------------------------------------------------------
-    ! 1. Initialize time
+    ! 1. Get ROF subcycling
     !-------------------------------------------------------
-    ! If routing time step dt [sec] and coupling time step coupling_period [day] is different, match dt to coupling_period.
-    if (dt/=coupling_period) then
+    delt_coupling = coupling_period*secprday   ! day -> sec
+
+    if (abs(mod(delt_coupling, dt)-0._r8)>verySmall) then
```

Comment:
I'm glad you are doing this checking or otherwise I'd suggest it. How big is verySmall in this case it likely should be zero, as you really need it to be exact.

Reply:
Hi Erik, verySmall=tiny(1.0_dp).   so smallest number resolved by double precision?

Reply:
replace `verySmall` with `0.0_dp`